### PR TITLE
[API-DRAFT]refactor api base

### PR DESCRIPTION
**Pull request by CalvinKirs**
Add `flink-core-starter` and `seatunnel-starter` to adapt to the new API

### CODE REVIEW

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/args/FlinkCommandArgs.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Parameter(names = {"-r", "--run-mode"},
+        converter = RunModeConverter.class,
+        description = "job run mode, run or run-application")
+    private FlinkRunMode runMode = FlinkRunMode.RUN;
+
+    @Parameter(names = {"-api", "--api-type"},
+        converter = ApiTypeConverter.class,
+        description = "Api type, engine or seatunnel")
+    private ApiType apiType = ApiType.ENGINE_API;
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/args/FlinkCommandArgs.java]: We can remove this parameter, in our new module we don't need this.

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/command/FlinkApiConfValidateCommand.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+
+/**
+ * Used to check the Flink conf is validated.
+ */
+public class FlinkApiConfValidateCommand implements Command<FlinkCommandArgs> {
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/command/FlinkApiConfValidateCommand.java:34]: This class it used in our old flink job, we don't need this in our new API module.

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/command/FlinkCommandBuilder.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            throw new IllegalArgumentException(
+                String.format("Deploy mode: %s is Illegal", commandArgs.getDeployMode()));
+        }
+        switch (commandArgs.getApiType()) {
+            case ENGINE_API:
+                return new FlinkApiCommandBuilder().buildCommand(commandArgs);
+            case SEATUNNEL_API:
+                return new SeaTunnelApiCommandBuilder().buildCommand(commandArgs);
+            default:
+                throw new IllegalArgumentException("Unsupported API type: " + commandArgs.getApiType());
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/command/FlinkCommandBuilder.java]: We also don't need to do this judge.

File: seatunnel-examples/seatunnel-flink-examples/src/main/java/org/apache/seatunnel/example/flink/LocalFlinkExample.java [hunk 4]
```diff
@@ -17,11 +17,11 @@
 
 package org.apache.seatunnel.example.flink;
 
-import org.apache.seatunnel.core.base.Seatunnel;
-import org.apache.seatunnel.core.base.command.Command;
-import org.apache.seatunnel.core.base.exception.CommandException;
-import org.apache.seatunnel.core.flink.args.FlinkCommandArgs;
-import org.apache.seatunnel.core.flink.command.FlinkCommandBuilder;
+import org.apache.seatunnel.core.starter.Seatunnel;
```
Comment (ruanwenjun) [inline seatunnel-examples/seatunnel-flink-examples/src/main/java/org/apache/seatunnel/example/flink/LocalFlinkExample.java]: The class `LocalFlinkExample` can be deleted in our new module.

**Review by ruanwenjun — COMMENTED** [review]
Here are some files need to be deleted, you can use another PR to polish.

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/args/FlinkCommandArgs.java (hunk 1 above)
Comment (CalvinKirs) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/args/FlinkCommandArgs.java]: thx~ done

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/command/FlinkApiConfValidateCommand.java (hunk 2 above)
Comment (CalvinKirs) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/command/FlinkApiConfValidateCommand.java:34]: thx~ done

**Pull request merged**